File: distribution/python/dqops/client/models/table_comparison_column_results_model.py
from typing import TYPE_CHECKING, Any, Dict, List, Type, TypeVar, Union

from attrs import define as _attrs_define
from attrs import field as _attrs_field

from ..types import UNSET, Unset

if TYPE_CHECKING:
    from ..models.table_comparison_column_results_model_column_comparison_results import (
        TableComparisonColumnResultsModelColumnComparisonResults,
    )


T = TypeVar("T", bound="TableComparisonColumnResultsModel")


@_attrs_define
class TableComparisonColumnResultsModel:
    """The table comparison column results model with the information about the most recent table comparison relating to a
    single compared column.

        Attributes:
            column_name (Union[Unset, str]): Column name
            column_comparison_results (Union[Unset, TableComparisonColumnResultsModelColumnComparisonResults]): The
                dictionary of comparison results between the tables for the specific column. The keys for the dictionary are
                check names. The values are summaries of the most recent comparison on this column.
    """

    column_name: Union[Unset, str] = UNSET
    column_comparison_results: Union[
        Unset, "TableComparisonColumnResultsModelColumnComparisonResults"
    ] = UNSET
    additional_properties: Dict[str, Any] = _attrs_field(init=False, factory=dict)

    def to_dict(self) -> Dict[str, Any]:
        column_name = self.column_name
        column_comparison_results: Union[Unset, Dict[str, Any]] = UNSET
        if not isinstance(self.column_comparison_results, Unset):
            column_comparison_results = self.column_comparison_results.to_dict()

        field_dict: Dict[str, Any] = {}
        field_dict.update(self.additional_properties)
        field_dict.update({})
        if column_name is not UNSET:
            field_dict["column_name"] = column_name
        if column_comparison_results is not UNSET:
            field_dict["column_comparison_results"] = column_comparison_results

        return field_dict

    @classmethod
    def from_dict(cls: Type[T], src_dict: Dict[str, Any]) -> T:
        from ..models.table_comparison_column_results_model_column_comparison_results import (
            TableComparisonColumnResultsModelColumnComparisonResults,
        )

        d = src_dict.copy()
        column_name = d.pop("column_name", UNSET)

        _column_comparison_results = d.pop("column_comparison_results", UNSET)
        column_comparison_results: Union[
            Unset, TableComparisonColumnResultsModelColumnComparisonResults
        ]
        if isinstance(_column_comparison_results, Unset):
            column_comparison_results = UNSET
        else:
            column_comparison_results = (
                TableComparisonColumnResultsModelColumnComparisonResults.from_dict(
                    _column_comparison_results
                )
            )

        table_comparison_column_results_model = cls(
            column_name=column_name,
            column_comparison_results=column_comparison_results,
        )

        table_comparison_column_results_model.additional_properties = d
        return table_comparison_column_results_model

    @property
    def additional_keys(self) -> List[str]:
        return list(self.additional_properties.keys())

    def __getitem__(self, key: str) -> Any:
        return self.additional_properties[key]

    def __setitem__(self, key: str, value: Any) -> None:
        self.additional_properties[key] = value

    def __delitem__(self, key: str) -> None:
        del self.additional_properties[key]

    def __contains__(self, key: str) -> bool:
        return key in self.additional_properties

File: dqops/src/main/java/com/dqops/data/checkresults/models/TableComparisonColumnResultsModel.java
/*
 * Copyright © 2021-Present DQOps, Documati sp. z o.o. (support@dqops.com)
 *
 * This file is licensed under the Business Source License 1.1,
 * which can be found in the root directory of this repository.
 *
 * Change Date: This file will be licensed under the Apache License, Version 2.0,
 * four (4) years from its last modification date.
 */
package com.dqops.data.checkresults.models;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonPropertyDescription;
import com.fasterxml.jackson.databind.PropertyNamingStrategies;
import com.fasterxml.jackson.databind.annotation.JsonNaming;
import io.swagger.annotations.ApiModel;
import lombok.Data;

import java.util.LinkedHashMap;
import java.util.Map;

/**
 * The table comparison column results model with the information about the most recent table comparison relating to a single compared column.
 */
@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonNaming(PropertyNamingStrategies.SnakeCaseStrategy.class)
@ApiModel(value = "TableComparisonColumnResultsModel", description = "The table comparison column results model with the information about the most recent table comparison relating to a single compared column.")
@Data
public class TableComparisonColumnResultsModel {
    /**
     * Column name.
     */
    @JsonPropertyDescription("Column name")
    private final String columnName;
    
    /**
     * The dictionary of comparison results between the tables for the specific column.
     * The keys for the dictionary are check names. The values are summaries of the most recent comparison on this column.
     */
    @JsonPropertyDescription("The dictionary of comparison results between the tables for the specific column. " +
            "The keys for the dictionary are check names. The values are summaries of the most recent comparison on this column.")
    private Map<String, ComparisonCheckResultModel> columnComparisonResults = new LinkedHashMap<>();

    public TableComparisonColumnResultsModel(String columnName) {
        this.columnName = columnName;
    }
}
